core/src/impl/Kokkos_Core.cpp

```diff
@@ -151,6 +155,31 @@ void combine(Kokkos::Tools::InitArguments& out,
   }
 }
 
+int get_device_count() {
+#if defined(KOKKOS_ENABLE_CUDA)
+  return Kokkos::Cuda::detect_device_count();
+#elif defined(KOKKOS_ENABLE_HIP)
+  return Kokkos::Experimental::HIP::detect_device_count();
+#elif defined(KOKKOS_ENABLE_SYCL)
+  return sycl::device::get_devices(sycl::info::device_type::gpu).size();
+#else
+  Kokkos::abort("implementation bug");
+  return -1;
```

Comment:
this could be a better message. Maybe 'Kokkos::get_device_count() not implemented for the enabled backend'